tool/src/js/structs.rs

```diff
@@ -478,7 +502,13 @@ pub fn gen_ts_type<W: fmt::Write>(
             ast::PrimitiveType::bool => write!(out, "Uint8Array")?,
             ast::PrimitiveType::char => write!(out, "Uint32Array")?,
         },
-        ast::TypeName::Unit => out.write_str("void")?,
+        ast::TypeName::Unit => {
+            if is_return_type {
+                out.write_str("void")?;
+            } else {
+                out.write_str("{}")?;
```

Comment:
Question: where does this `{}` get used?

Reply:
Methods that return `Result<T, ()>` use this because when we write the `@throws` thing, I wanted to specify the content of the `FFIError`. This lead to it generating `@throws FFIError<void>`, while the JS code is actually throwing `FFIError<{}>`. Should I change this to a more explicit enum, or at least change the name to `is_ffierror_value` and invert the usage?

Reply:
You should be able to set `error_value` to `undefined` and then treat it as an `FFIError<void>`